src/tools/miri/src/helpers.rs

```diff
@@ -915,12 +919,25 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     }
 
     /// Check that the ABI is what we expect.
-    fn check_abi<'a>(&self, abi: ExternAbi, exp_abi: ExternAbi) -> InterpResult<'a, ()> {
-        if abi != exp_abi {
+    fn check_abi<'a>(&self, fn_abi: &FnAbi<'tcx, Ty<'tcx>>, exp_abi: Conv) -> InterpResult<'a, ()> {
+        let caller_abi_name = match fn_abi.conv {
+            Conv::C => "C",
+            Conv::Rust => "Rust",
+            Conv::X86Stdcall => "X86Stdcall",
+            _ => { panic!("Unsupported caller ABI");}
+        }; 
+        let callee_abi_name = match exp_abi {
+            Conv::C => "C",
+            Conv::Rust => "Rust",
+            Conv::X86Stdcall => "X86Stdcall",
+            _ => { panic!("Unsupported callee ABI");}
+        }; 
+
+        if fn_abi.conv != exp_abi {
             throw_ub_format!(
                 "calling a function with ABI {} using caller ABI {}",
```

Comment:
Just use debug formatting for the error for now. We do the same in `interpret/call.rs` already.

The proper fix here would be a `impl Display for Conv` somewhere near [this](https://github.com/rust-lang/rust/blob/9aab517d6310223ac5a89c640723a64b695d49d2/compiler/rustc_target/src/callconv/mod.rs#L849-L852). If you want you can do that in a future PR. :)

Reply:
Sure, I can open one after this.